share/functions/__fish_config_interactive.fish

```diff
@@ -254,4 +254,62 @@ function __fish_config_interactive -d "Initializations that should be performed
 		end
 		__fish_command_not_found_handler $argv
 	end
+	if test $TERM = "linux" # A linux in-kernel VT with 8 colors and 256/512 glyphs
+		# In a VT we have
+		# black (invisible)
+		# red
+		# green
+		# yellow
+		# blue
+		# magenta
+		# cyan
+		# white
+
+		# Pretty much just set at random
+		set -g fish_color_normal normal
+		set -g fish_color_command yellow
+		set -g fish_color_param cyan
+		set -g fish_color_redirection normal
+		set -g fish_color_comment red
+		set -g fish_color_error red
+		set -g fish_color_escape cyan
+		set -g fish_color_operator cyan
+		set -g fish_color_quote blue
+		set -g fish_color_autosuggestion yellow
+		set -g fish_color_valid_path
+		set -g fish_color_cwd green
+		set -g fish_color_cwd_root red
+		set -g fish_color_match cyan
+		set -g fish_color_history_current cyan
+		set -g fish_color_search_match cyan
+		set -g fish_color_selection blue
+		set -g fish_color_end yellow
+		set -g fish_color_host normal
+		set -g fish_color_status red
+		set -g fish_color_user green
+		set -g fish_pager_color_prefix cyan
+		set -g fish_pager_color_completion normal
+		set -g fish_pager_color_description yellow
+		set -g fish_pager_color_progress cyan
+
+		# Don't allow setting color other than what linux offers (see #2001)
+		functions -e set_color
+		function set_color
+			set -l term_colors black red green yellow blue magenta cyan white normal
```

Comment:
Sorry for the late review, but I'm not sure this is actually required; `set_color` should detect a non-256-colour terminal and adjust behaviour accordingly - although I haven't tested it so I might well be wrong.